libs/blocks/table/table.js

```diff
@@ -349,9 +366,17 @@ function applyStylesBasedOnScreenSize(table, originTable) {
           });
         }
       });
+      setRowStyle();
     };
 
-    if (!table.parentElement.querySelector('.filters')) {
+    // Remove filter if table there are only 2 columns
+    let filter = true;
+    if (isMerch && headingsLength <= 2) {
+      filter = false;
+    } else if (!isMerch && headingsLength <= 3) {
+      filter = false;
+    }
```

Comment:
Realized this can be simplified to:
`const filter = isMerch ? headingsLength > 2 : headingsLength > 3;`

Reply:
done